sdk/src/sealance/merkletree-support.ts

```diff
@@ -0,0 +1,237 @@
+        const bytes = bech32m.fromWords(words);
+
+        // Convert bytes to BigInt (little-endian)
+        let fieldValue = BigInt(0);
+        for (let i = 0; i < bytes.length; i++) {
+            fieldValue |= BigInt(bytes[i]) << BigInt(i * 8);
+        }
+
+        return fieldValue;
+    }
+
+    /**
+    * Hashes two elements using Poseidon4 hash function
+    * @param prefix - Prefix for the hash (e.g., "0field" for nodes, "1field" for leaves)
+    * @param el1 - First element to hash
+    * @param el2 - Second element to hash
+    * @returns The hash result as a Field
+    * @throws {Error} If inputs are empty or invalid
+    */
+    hashTwoElements(prefix: string, el1: string, el2: string): Field {
+        if (!el1 || !el2) {
+            throw new Error("Invalid inputs: elements cannot be empty");
+        }
+        const fields = [Field.fromString(prefix), Field.fromString(el1), Field.fromString(el2)];
+        const arrayPlaintext = Plaintext.fromString(`[${fields.map(f => f.toString()).join(",")}]`);
+
+        return Sealance.hasher.hash(arrayPlaintext.toFields());
+    }
+
+    /**
+    * Builds a Merkle tree from given leaves
+    * The tree is built bottom-up, hashing pairs of elements at each level
+    *
+    * @param leaves - Array of leaf elements (must have even number of elements)
+    * @returns Array representing the complete Merkle tree as BigInts
+    * @throws {Error} If leaves array is empty or has odd number of elements
+    *
+    * @example
+    * ```typescript
+    * const leaves = ["0field", "1field", "2field", "3field"];
+    * const tree = buildTree(leaves);
+    * const root = tree[tree.length - 1]; // Get the Merkle root
+    * ```
+    */
+    buildTree(leaves: string[]): bigint[] {
+        if (leaves.length === 0) {
+            throw new Error("Leaves array cannot be empty");
+        }
+        if (leaves.length % 2 !== 0) {
+            throw new Error("Leaves array must have even number of elements");
+        }
+        let currentLevel = leaves;
+        let tree = [...currentLevel];
+        let levelSize = currentLevel.length;
+
+        while (levelSize > 1) {
+            const nextLevel = [];
+            for (let i = 0; i < levelSize; i += 2) {
+                const left = currentLevel[i];
+                const right = currentLevel[i + 1];
+                const prefix = leaves.length === levelSize ? "1field" : "0field";
+                const hash = this.hashTwoElements(prefix, left, right);
+                nextLevel.push(hash.toString());
+            }
+            tree = [...tree, ...nextLevel];
+            currentLevel = nextLevel;
+            levelSize = currentLevel.length;
+        }
+        return tree.map(element => BigInt(element.slice(0, element.length - "field".length)));
+    }
+
+    /**
+    * Converts Leo addresses to field elements, sorts them, pads with zero fields, and returns an array
+    * This prepares addresses for Merkle tree construction
```

Comment:
Formatted to properly show up in JSDoc.

```suggestion
    * Converts Aleo addresses to field elements, sorts them, pads with zero fields, and returns an array. This prepares addresses for Merkle tree construction.
```